stl/inc/expected

```diff
@@ -0,0 +1,1034 @@
+        }
+    };
+
+    // clang-format off
+    expected(expected&&) requires
+        is_trivially_move_constructible_v<_Ty> && is_trivially_move_constructible_v<_Err> = default;
+    // clang-format on
+
+    template <class _Uty, class _UErr>
+    static constexpr bool _Is_not_convertible = !is_constructible_v<_Ty, expected<_Uty, _UErr>&> //
+                                             && !is_constructible_v<_Ty, expected<_Uty, _UErr>> //
+                                             && !is_constructible_v<_Ty, const expected<_Uty, _UErr>&> //
+                                             && !is_constructible_v<_Ty, const expected<_Uty, _UErr>> //
+                                             && !is_convertible_v<expected<_Uty, _UErr>&, _Ty> //
+                                             && !is_convertible_v<expected<_Uty, _UErr>&&, _Ty> //
+                                             && !is_convertible_v<const expected<_Uty, _UErr>&, _Ty> //
+                                             && !is_convertible_v<const expected<_Uty, _UErr>&&, _Ty> //
+                                             && !is_constructible_v<unexpected<_Err>, expected<_Uty, _UErr>&> //
+                                             && !is_constructible_v<unexpected<_Err>, expected<_Uty, _UErr>> //
+                                             && !is_constructible_v<unexpected<_Err>, const expected<_Uty, _UErr>&> //
+                                             && !is_constructible_v<unexpected<_Err>, const expected<_Uty, _UErr>>;
+
+    template <class _Uty, class _UErr>
+        requires is_constructible_v<_Ty, const _Uty&> && is_constructible_v<_Err, const _UErr&> //
+            && _Is_not_convertible<_Uty, _UErr>
+    constexpr explicit(!is_convertible_v<const _Uty&, _Ty> || !is_convertible_v<const _UErr&, _Err>)
+        expected(const expected<_Uty, _UErr>& _Other) noexcept(is_nothrow_constructible_v<_Ty, const _Uty&>&&
+                is_nothrow_constructible_v<_Err, const _UErr&>) // strengthened
+        : _Has_value(_Other._Has_value) {
+        if (_Has_value) {
+            _STD construct_at(_STD addressof(_Value), _Other._Value);
+        } else {
+            _STD construct_at(_STD addressof(_Unexpected), _Other._Unexpected);
+        }
+    }
+
+    template <class _Uty, class _UErr>
+        requires is_constructible_v<_Ty, _Uty> && is_constructible_v<_Err, _UErr> && _Is_not_convertible<_Uty, _UErr>
+    constexpr explicit(!is_convertible_v<_Uty, _Ty> || !is_convertible_v<_UErr, _Err>)
+        expected(expected<_Uty, _UErr>&& _Other) noexcept(
+            is_nothrow_constructible_v<_Ty, _Uty>&& is_nothrow_constructible_v<_Err, _UErr>) // strengthened
+        : _Has_value(_Other._Has_value) {
+        if (_Has_value) {
+            _STD construct_at(_STD addressof(_Value), _STD forward<_Uty>(_Other._Value));
+        } else {
+            _STD construct_at(_STD addressof(_Unexpected), _STD forward<_UErr>(_Other._Unexpected));

```

Comment:
@CaseyCarter notes that although this technically mirrors the Standardese, the Standard is taking a shortcut by specifying the converting copy and converting move ctors together (with `UF` and `GF`). `expected` has already forbidden reference types, so when we know we're implementing the converting move ctor, we can simply say `_STD move`. (We conventionally reserve `_STD forward` in the implementation only for perfect forwarding, or situations like `pair` where handling reference types is necessary.)

Reply:
I will never not be amazed, that @CaseyCarter is so good that he reads this monster of a function and still finds these minutiae